crates/bevy_input_focus/src/tab_navigation.rs

```diff
@@ -144,19 +180,29 @@ impl TabNavigation<'_, '_> {
                 })
         });
 
-        if focus.0.is_some() && tabgroup.is_none() {
-            warn!("No tab group found for focus entity. Users will not be able to navigate back to this entity.");
+        let navigation_result = self.navigate_in_group(tabgroup, focus, action);
+
+        match navigation_result {
+            Ok(entity) => {
+                if focus.0.is_some() && tabgroup.is_none() {
+                    Err(TabNavigationError::NoTabGroupForCurrentFocus {
+                        previous_focus: focus.0.unwrap(),
+                        new_focus: entity,
+                    })
+                } else {
+                    Ok(entity)
+                }
+            }
+            Err(e) => Err(e),
         }
```

Comment:
```suggestion
        self.navigate_in_group(tabgroup, focus, action)
            .and_then(|entity| {
                if focus.0.is_some() && tabgroup.is_none() {
                    Err(TabNavigationError::NoTabGroupForCurrentFocus {
                        previous_focus: focus.0.unwrap(),
                        new_focus: entity,
                    })
                } else {
                    Ok(entity)
                }
            })
```

Reply:
I've become obsessed with rewriting everything using chained combinators everywhere for some reason today, just ignore me if you hate this style.

Reply:
Ha, I prefer the nested matching / if's: it's easier for me to piece together what's going on without remembering what all of the little methods do.

Reply:
Maybe I'm missing something, but couldn't _both_ nesting _and_ combinators be prevented here with a regular ol `?`? Also swapped `is_some` + `unwrap` with `if let`, which reintroduces one level of nesting until let-chains are available I guess.

Also same disclaimer, feel free to ignore ^^

```suggestion
        let entity = self.navigate_in_group(tabgroup, focus, action)?;

        if let Some(previous_focus) = focus.0 {
            if tabgroup.is_none() {
                return Err(TabNavigationError::NoTabGroupForCurrentFocus {
                    previous_focus,
                    new_focus: entity,
                });
            }
        }

        Ok(entity)
```